### PR TITLE
[WIRE-419] Update iot-client to latest generated artifact

### PR DESCRIPTION
**Pull request by rjtokenring**
### Motivation
Upgrade iot-client to new client structure due to changes in code generation done by openapi-generator

### Reviewer checklist

* [ ] PR address a single concern.
* [ ] PR title and description are properly filled.
* [ ] Changes will be merged in `main`.
* [ ] Changes are covered by tests.
* [ ] Logging is meaningful in case of troubleshooting.
* [ ] History is clean, commit messages are meaningful (see `CONTRIBUTING.md`) and are well formatted.

### CODE REVIEW

**Review by robgee86 — APPROVED** [review]

File: .licenses/go/github.com/arduino/iot-client-go.dep.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
 name: github.com/arduino/iot-client-go
-version: v1.4.4
+version: v1.4.5-0.20240416140423-fb8fda8a47af
```
[the remark this answers is not in the document]
Comment (rjtokenring) [reply .licenses/go/github.com/arduino/iot-client-go.dep.yml:3]: Problem is that iot-client-go project structure has not yet been ported to a fully go compliant go v2 module structure.
So, for now, we need to pin directly the hash commit. It is in plan to fix module structure.

File: go.mod [hunk 2]
```diff
@@ -7,10 +7,11 @@ require (
 	github.com/arduino/arduino-cli v0.0.0-20221116144942-76251df9241a
 	github.com/arduino/go-paths-helper v1.7.0
 	github.com/arduino/go-win32-utils v1.0.0
-	github.com/arduino/iot-client-go v1.4.4
+	github.com/arduino/iot-client-go v1.4.5-0.20240416140423-fb8fda8a47af
```
[the remark this answers is not in the document]
Comment (rjtokenring) [reply go.mod:10]: see above

File: .licenses/go/github.com/arduino/iot-client-go.dep.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (rjtokenring) [reply .licenses/go/github.com/arduino/iot-client-go.dep.yml]: Release and fixed iot-client-go project. Now it is possible to pull as s std /v2 module.
PR updated accordingly

**Review by robgee86 — APPROVED** [review]

**Review by mirkokurt — COMMENTED** [review]
:arduinolover:

**Pull request merged**